src/NuGet.Core/NuGet.Frameworks/comparers/NuGetFrameworkFullComparer.cs

```diff
@@ -32,7 +32,8 @@ public bool Equals(NuGetFramework x, NuGetFramework y)
 
             return x.Version == y.Version
                    && StringComparer.OrdinalIgnoreCase.Equals(x.Framework, y.Framework)
-                   && StringComparer.OrdinalIgnoreCase.Equals(x.Profile, y.Profile);
+                   && StringComparer.OrdinalIgnoreCase.Equals(x.Profile, y.Profile)
+                   && !x.IsInvalid;
```

Comment:
Check !IsInvalid first (move it up a couple lines) so we avoid doing the Equals when it's `true`.

Reply:
in this case, invalid is going to be very rare - since it will never work, and will be failing. i'd rather keep the success/failure check ignoring invalid. please comment if i'm missing something or if you want to discuss more.